packages/swapper/src/swappers/zrx/zrxExecuteTrade/zrxExecuteTrade.ts

```diff
@@ -15,7 +15,7 @@ export async function zrxExecuteTrade(
     const value = sellAsset.assetId === 'eip155:1/slip44:60' ? trade.sellAmount : '0'
     const adapter = await adapterManager.byChainId(sellAsset.chainId)
     const bip44Params = adapter.buildBIP44Params({
-      accountNumber: bnOrZero(trade.sellAssetAccountId).toNumber()
+      accountNumber: bnOrZero(trade.sellAssetAccountNumber).toNumber()
```

Comment:
Do we still need to `bnOrZero().toNumber()` this since it should already be a number according to `trade` typings?

Reply:
> Do we still need to `bnOrZero().toNumber()` this since it should already be a number according to `trade` typings?

You have to be careful about that if `trade` is the result from third-party code like an API/Web3 call. There's no runtime type checking.

We also need to ask ourselves if defaulting to account #0 is the right thing to do in this case or should we throw an error if `accountNumber` is not value?